### PR TITLE
feat: build qa for login/register/you/profile/edit

### PR DESCRIPTION
**Pull request by webdeveloper001**
### Ticket
https://www.notion.so/thegoodparty/

### CODE REVIEW

File: cypress/integration/you/edit.spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  //   });
+  // });
+  // it('check election section without zipcode', () => {
+  //   cy.checkElectionSectionInYou(
+  //     user,
+  //     senateCandidatesCount,
+  //     houseCandidatesCount,
+  //     getRankingObj(ranking),
+  //   );
+  // });
```
Comment (tomer-tgp) [inline cypress/integration/you/edit.spec.js:40]: do we need this commented code?

File: cypress/integration/you/you.spec.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      );
+    });
+    Object.keys(testZipcodes).forEach(zipcode => {
+      it(`check election section - ${zipcode} ${testZipcodes[zipcode]}`, () => {
+        cy.chooseCorrectZipcode(zipcode);
+        cy.visit('/you');
+        cy.getCookie('user').then(cookie => {
+          user = parseCookie(cookie.value);
+          districtNumber = userDistrict(user);
+          console.log('parseduser', user);
```
Comment (tomer-tgp) [inline cypress/integration/you/you.spec.js:59]: remove logs

File: cypress/integration/you/you.spec.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        senateCandidatesCount,
+        houseCandidatesCount,
+        getRankingObj(ranking),
+      );
+    });
+    Object.keys(testZipcodes).forEach(zipcode => {
+      it(`check election section - ${zipcode} ${testZipcodes[zipcode]}`, () => {
+        cy.chooseCorrectZipcode(zipcode);
+        cy.visit('/you');
+        cy.getCookie('user').then(cookie => {
```
Comment (tomer-tgp) [inline cypress/integration/you/you.spec.js:56]: can we not use async await in stead of callbacks?

File: cypress/support/commands/auth.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+Cypress.Commands.add('checkEmailRegisterSection', () => {
+  cy.get('[data-cy=email-register]')
+    .should('contain', 'CONTINUE WITH EMAIL')
+    .should('have.attr', 'href')
+    .and('contain', '/you/register-email');
+  cy.get('[data-cy=email-register]').click();
+  cy.checkEmailRegisterPage();
+});
+Cypress.Commands.add('checkLoginConfirmPage', email => {
+  cy.wait(2000);
```
Comment (tomer-tgp) [inline cypress/support/commands/auth.js:37]: wait(2000) will cause flaky tests. 
please read
https://docs.cypress.io/guides/references/best-practices.html#Unnecessary-Waiting

File: cypress/support/commands/you.js [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+    .should('contain', 'Invite 3 or more friends to join,')
+    .and('contain', 'and watch how quickly The Good Party');
+  cy.get('[data-cy=invite-link-label]').should(
+    'contain',
+    'Your Unique Invite Link',
+  );
+  cy.get('[data-cy=invite-url]').should('contain', url);
+});
+
+Cypress.Commands.add('checkHelpSectionInYou', () => {
```
Comment (tomer-tgp) [inline cypress/support/commands/you.js:169]: Should these be commands or just tests?
My understanding is that commands should be reusable like "login".

File: package.json [hunk 6]
```diff
@@ -83,6 +83,7 @@
     "compression": "1.7.4",
     "connected-react-router": "6.4.0",
     "cross-env": "5.2.0",
+    "cypress-localstorage-commands": "^1.2.1",
```
Comment (tomer-tgp) [inline package.json:86]: we are not using localstorage, I saw you commented out the code that imports this. Should we remove the package?

File: cypress/integration/you/you.spec.js (hunk 3 above)
Comment (webdeveloper001) [reply cypress/integration/you/you.spec.js:56]: I tried but async await was not working properly during the test.

File: cypress/support/commands/auth.js (hunk 4 above)
Comment (tomer-tgp) [reply cypress/support/commands/auth.js]: still an issue. Can we avoid waits?

File: cypress/integration/auth/login.spec.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+describe('Login', () => {
+  it('finds correct text', () => {
+    cy.visit('/login');
+    cy.get('[data-cy=page-title]').contains('Sign into your account');
+    cy.get('[data-cy=title]').contains('Sign into your account');
+    cy.get('[data-cy=register-label]').contains('t have an account?');
+  });
+  it('finds social login part', () => {
+    cy.wait(3000);
```
Comment (tomer-tgp) [inline cypress/integration/auth/login.spec.js:11]: Can we avoid waits?

File: cypress/support/commands/you.js [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+    .should('have.attr', 'href')
+    .and('contain', '?register=true');
+});
+Cypress.Commands.add('checkInitialInfoEditSection', () => {
+  cy.get('[data-cy=new-name]').type('blueshark0811');
+  cy.get('[data-cy=new-feedback]').type('The Good Party');
+  cy.get('[data-cy=profile-form]')
+    .find('button')
+    .click();
+  cy.wait(4000);
```
Comment (tomer-tgp) [inline cypress/support/commands/you.js:210]: Can we avoid waits?

**Pull request merged**